system-int-test/test/cmr/system_int_test/bootstrap/bulk_index/generics_test.clj

```diff
@@ -0,0 +1,178 @@
+
+     (is (= 0 (:hits (search/find-refs :grid {}))))
+
+     (bootstrap/bulk-index-grids)
+     (index/wait-until-indexed)
+     
+     (testing "Grid concepts are indexed." 
+       (let [{:keys [hits refs]} (search/find-refs :grid {})] 
+         (is (= 3 hits)) 
+         (is (= 3 (count refs)))))
+     ;; Re-enable message publishing.
+     (core/reenable-automatic-indexing))))
+
+(deftest ^:oracle bulk-index-grid-revisions
+  (testing "Bulk index grids index all revisions index as well"
+    (system/only-with-real-database
+     ;; Disable message publishing so items are not indexed.
+     (core/disable-automatic-indexing)
+     (let [token (echo-util/login (system/context) "user1")
+           grid1-1 (generic/ingest-generic-document nil "PROV1" "Test-Native-Id-1"
+                                                    :grid generic/grid-good :post)
+           grid1-2-tombstone (merge (generic/ingest-generic-document token "PROV1" "Test-Native-Id-1"
+                                                                     :grid generic/grid-good :delete)
+                                    generic/grid-good
+                                    {:deleted true
+                                     :user-id "user1"})
+           grid1-3 (generic/ingest-generic-document nil "PROV1" "Test-Native-Id-1"
+                                                    :grid generic/grid-good :put)
+           
+           grid2-1 (generic/ingest-generic-document nil "PROV2" "Test-Native-Id-2"
+                                                    :grid generic/grid-good :post)
+           grid2-2 (generic/ingest-generic-document nil "PROV2" "Test-Native-Id-2"
+                                                    :grid generic/grid-good :put)
+           grid2-3-tombstone (merge (generic/ingest-generic-document token "PROV2" "Test-Native-Id-2"
+                                                                     :grid generic/grid-good :delete)
+                                    generic/grid-good
+                                    {:deleted true
+                                     :user-id "user1"})
+           
+           grid3 (generic/ingest-generic-document nil "PROV3" "Test-Native-Id-3"
+                                                  :grid generic/grid-good :post)]
+
+       (testing "Before bulk indexing, search for grids found nothing"
+         (let [{:keys [status results]} (search/find-concepts-umm-json :grid {:all-revisions true})]
+           (is (= 200 status))
+           (is (= 0 (:hits results)))))
+
+       ;; Just index PROV1
+       (bootstrap/bulk-index-grids "PROV1")
+       (index/wait-until-indexed)
+
+       (testing "After bulk indexing a provider, search found all grid revisions for that provider"
+         (let [{:keys [status results]} (search/find-concepts-umm-json :grid {:all-revisions true})]
+           (is (= 200 status))
+           (is (= 3 (:hits results)))))
```

Comment:
Again, we should add checking of concept ids (at least, preferably revision-ids) to be thorough.